includes/IslandoraHandleHandleHandler.class.inc

```diff
@@ -52,17 +52,17 @@ abstract class IslandoraHandleHandleHandler {
   /**
    * Helper to construct a target URL.
    *
-   * Constructed from islandora/object, appended with the given $suffix and
+   * Constructed from islandora/object, appended with the given $object->id and
    * prepended with either the local server's URL or the custom configured
    * variable islandora_handle_host.
    *
-   * @param string $suffix
-   *   Suffix to append to the URL.
+   * @param AbstractObject $object
+   *   The object that is the target of the URL.
    *
    * @return string
    *   The constructed target URL.
    */
-  public function constructTargetUrl($suffix) {
+  public function constructTargetUrl(AbstractObject $object) {
```

Comment:
For the sake of backwards compatibility,  changing the function signature here doesn't seem like the right solution. We could keep the `string` argument type and have callers pass `$object->id` instead of the object itself - seeing as this function is only making use of the PID anyway. Did you have a particular use case in mind for this change?

Reply:
If the `$object` itself is needed, perhaps it could just be loaded from the PID? That way the function signature wouldn't need to change. Or the signature could be changed to accept _either_ the `string` ID or the `AbstractObject` object?

Reply:
I see your point. My reasoning was that constructTargetUrl makes a target url for an object, so we should supply it with an object. How this function then makes the url from that object, is an implementation detail. Also, using the suffix of the handle is not right, because constructSuffix now returns the pid of the object, but this might change in the future (as commented in that function).
But I see your point about backwards compatibility, so I will make the change that the function accepts either a string pid or the object.

Reply:
Right, your reasoning certainly makes sense - I think this change is a nice compromise.